src/main/java/com/externalSorting/app/ExternalSortingRunner.java

```diff
@@ -1,24 +1,44 @@
 package com.externalSorting.app;
 
 import java.io.*;
-import java.nio.file.Files;
-import java.nio.file.Paths;
 import java.util.*;
 
 
-public class ExternalSorting {
-    /*
+public class ExternalSortingRunner {
+
+    private final String originalFile;
+    private final String outputFolder;
+
+    public ExternalSortingRunner(String originalFile, String outputFile) {
+        this.originalFile = originalFile;
+        this.outputFolder = outputFile;
+    }
+
+    protected void externalSort() {
+
+        System.out.println(outputFolder + "/tmpdir");
+        File newDirectory = new File("src/main/resources/tmpdir/");
+        List<File> files = readAndSplitData("src/main/resources/SmallLibrary.txt", newDirectory);
+
+        System.out.println(outputFolder + "/Output.txt");
+        mergeFiles(files, new File("src/main/resources/SmallLibraryOutput.txt"), comparator);
+        for (File file : files) {
+            file.delete();
+        }
+        newDirectory.delete();
+    }
+    /**
     CHANGE_FILE
     Maximum number of lines of text that can be put into a single temporary file
     */
-    private static final int CHANGE_FILE = 400000;
+    private static final int TMP_FILE_MAXSIZE = 24000000; // 24mb
 
-    /*
+    /**
     Defines a default string comparator
     */
     private static final Comparator<String> comparator = (o1, o2) -> o1.compareTo(o2);
```

Comment:
Can be replaced by Comparator.naturalOrder()

Sometimes fragments of code are highlighted in yellow - that means that the compiler has some useful hints for you. It's worth looking at them.